app/pages/lab/workflow.jsx

```diff
@@ -0,0 +1,1042 @@
+          </ModalFormDialog> : undefined}
+        <p className="form-help">A workflow is the sequence of tasks that you’re asking volunteers to perform. For example, you might want to ask volunteers to answer questions about your images, or to mark features in your images, or both.</p>
+        {this.props.project.live && this.props.workflow.active ?
+          <p className="form-help warning">
+            <strong>
+              You cannot edit tasks for an active workflow on a live project, all the task fields below will be inactive.
+              <br />
+              To edit these fields, deactivate your workflow on the <Link to={`/lab/${this.props.project.id}/workflows`}>workflows page</Link> - do not change your project to development.
+            </strong>
+          </p> : undefined}
+        <div className="columns-container">
+          <div className="column">
+            <div>
+              <AutoSave tag="label" resource={this.props.workflow}>
+                <label className="form-label" for="displayName">Workflow title</label>
+                <br />
+                <input type="text" id="displayName" name="display_name" value={this.props.workflow.display_name} className="standard-input full" onChange={handleInputChange.bind(this.props.workflow)} />
+              </AutoSave>
+              <small className="form-help">If you let your volunteers choose which workflow to attempt, this text will appear as an option on the project front page.</small>
+
+              <br />
+
+              <div className={disabledIfLive}>
+                <div className="nav-list standalone">
+                  <span className="nav-list-header">Tasks</span>
+                  <br />
+                  {noTasksDefined ?
+                    <div className="nav-list-item">
+                      <small className="form-help">(No tasks yet)</small>
+                    </div>
+                  :
+                    (() => {
+                      const result = [];
+                      for (let key in this.props.workflow.tasks) {
+                        definition = this.props.workflow.tasks[key];
+                        if (definition.type !== 'shortcut') {
+                          const classNames = ['secret-button', 'nav-list-item'];
+                          const taskDefinition = taskComponents[definition.type]?.getTaskText(definition);
+                          if (key === this.state.selectedTaskKey) {
+                            classNames.push('active');
+                          }
+                          result.push(<div key={key}>
+                            <button
+                              type="button"
+                              className={classNames.join(' ')}
+                              onClick={this.setState.bind(this, {selectedTaskKey: key}, null)}
+                            >
+                              {(() => { switch (definition.type) {
```

Comment:
Another auto-running anonymous function, nested inside the first auto running anonymous function. I think this is going to be really hard to read and debug.

Reply:
I think this function could be extracted from the component and declared at the top of the file, which would make the code much easier to understand.